lib/modules/manager/npm/update/dependency/index.ts

```diff
@@ -25,7 +28,18 @@ function replaceAsString(
   }
   // Look for the old version number
   const searchString = `"${oldValue}"`;
-  const newString = `"${newValue}"`;
+  let newString = `"${newValue}"`;
+
+  if (patchReg.test(oldValue)) {
+    const replaceRegex = regEx(`(patch:${depName}@(npm:)?).*#`);
+    const match = patchReg.exec(oldValue);
+    const patch = oldValue.replace(replaceRegex, `${match[1]}${newValue}#`);
+    if (patch) {
+      parsedContents[depType][depName] = patch;
+      newString = `"${patch}"`;
+    }
+  }
```

Comment:
depName cant be part of the regex testing, as it will cause regression on other tests,
the value of depName could be anything which could cause `Invalid Regex` and fail

Reply:
on the other hand i removed the check for if patch is falsey

Reply:
the main issue i have with 2 regexes is that :
1) i need the detection regex to make sure we are in "patch"
2) i need the replace regex, which replaces only the version part of the regex

you are trying to combine both, but it will cause other issues, its safer this way, 
if you have a better way without ruining other tests please suggest